src/context.c

```diff
@@ -571,6 +571,51 @@ lysc_node_clear_priv_dfs_cb(struct lysc_node *node, void *UNUSED(data), ly_bool
     return LY_SUCCESS;
 }
 
+static void
+lysc_node_clear_all_priv(struct lys_module *mod)
+{
+    LY_ARRAY_COUNT_TYPE u, v;
+    /* lys_compile_extension_instance() was called */
+    ly_bool lcei_called = 0;
+
+    if (mod->compiled) {
+        /* set NULL for all ::lysc_node.priv pointers in module */
+        lysc_module_dfs_full(mod, lysc_node_clear_priv_dfs_cb, NULL);
+    }
+
+    /* only lys_compile_extension_instance()
+     * can set ::lysp_ext_instance.parsed
+     */
+    if (mod->parsed) {
+        struct lysp_ext_instance *exts_p;
+        exts_p = mod->parsed->exts;
+        LY_ARRAY_FOR(exts_p, u) {
+            if (exts_p[u].parsed) {
+                lcei_called = 1;
+                break;
+            }
+        }
+    }
+
+    if (lcei_called) {
+        struct lysc_ext_instance *exts;
+        struct lysc_ext_substmt *substmts;
+        struct lysc_node *root;
+
+        /* set NULL for all ::lysc_node.priv pointers in extensions */
+        exts = mod->compiled->exts;
+        LY_ARRAY_FOR(exts, u) {
+            substmts = exts[u].substmts;
+            LY_ARRAY_FOR(substmts, v) {
+                if (LY_STMT_IS_DATA_NODE(substmts[v].stmt)) {
+                    root = *(struct lysc_node **)substmts[v].storage;
+                    lysc_tree_dfs_full(root, lysc_node_clear_priv_dfs_cb, NULL);
+                }
+            }
+        }
+    }
```

Comment:
This still does not seem safe to me. Just because you have found a single parsed extension that has the compiled nodes, you treat all of them that way? It must be specific for each parsed extension instance.